packages/authorization/src/decorators/authorize.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/authorization
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  MetadataInspector,
+  Constructor,
+  MethodDecoratorFactory,
+} from '@loopback/context';
+import {AuthorizationBindings} from '../keys';
+
+/**
+ * Authorization metadata stored via Reflection API
+ */
+export interface AuthorizationMetadata {
+  allow?: string[];
+  deny?: string[];
+  scopes?: string[];
+}
+
+/**
+ * Mark a controller method as requiring authorized user.
+ *
+ * @param spec Authorization metadata
+ */
+export function authorize(spec: AuthorizationMetadata) {
+  return MethodDecoratorFactory.createDecorator<AuthorizationMetadata>(
+    AuthorizationBindings.METADATA,
+    spec,
+  );
+}
+
+export namespace authorize {
+  export const allowAll = () => authorize({allow: ['$everyone']});
+  export const allowAuthenticated = () =>
+    authorize({allow: ['$authenticated']});
+  export const denyAll = () => authorize({deny: ['$everyone']});
+  export const denyUnauthenticated = () =>
+    authorize({deny: ['$anonymous', 'unauthenticated']});
```

Comment:
`$unauthenticated`?

Reply:
What is the difference between `$anonymous` and `$unauthenticated` ?

Reply:
One must be careful while implementing the _all_ roles. Very often what the developer wants is _all but_ . A developer may be tempted to write something like : 

```typescript
@authorize({allow: ['MODERATOR']})
@authorize.denyAll()
// maybe a helper like this may be wanted
@authorize.denyAllBut({allow: ['MODERATOR']})
// or of course
@authorize.allowAllBut({allow: ['$anonymous']})
myMethod();
```

This is a case to handle. The order of annotation is very important and counter intuitive so the documentation must warn the user about that. Hence the helper `denyAllBut();`

Reply:
There should be a rule that makes the order of decorators irrelevant.